pkg/apply/task/apply_task.go

```diff
@@ -115,29 +108,29 @@ func (a *ApplyTask) Start(taskContext *taskrunner.TaskContext) {
 				taskContext.CaptureResourceFailure(id)
 				continue
 			}
-			clusterObj, err := getClusterObj(dynamic, info)
-			if err != nil {
-				if !apierrors.IsNotFound(err) {
-					if klog.V(4).Enabled() {
-						klog.Errorf("error (%s) retrieving %s/%s from cluster--continue",
-							err, info.Namespace, info.Name)
+			// Check filters to see if we're prevented from applying.
+			var filtered bool
+			var filterErr error
+			for _, filter := range a.Filters {
+				klog.V(6).Infof("apply filter %s: %s", filter.Name(), id)
+				var reason string
+				filtered, reason, filterErr = filter.Filter(obj)
+				if filterErr != nil {
+					if klog.V(5).Enabled() {
+						klog.Errorf("error during %s, (%s): %s", filter.Name(), id, filterErr)
 					}
-					taskContext.EventChannel() <- createApplyFailedEvent(id, err)
+					taskContext.EventChannel() <- createApplyFailedEvent(id, filterErr)
 					taskContext.CaptureResourceFailure(id)
-					continue
+					break
 				}
-			}
-			canApply, err := inventory.CanApply(a.InvInfo, clusterObj, a.InventoryPolicy)
-			if !canApply {
-				klog.V(5).Infof("can not apply %s/%s--continue",
-					clusterObj.GetNamespace(), clusterObj.GetName())
-				if err != nil {
-					taskContext.EventChannel() <- createApplyFailedEvent(id, err)
-				} else {
-					taskContext.EventChannel() <- createApplyEvent(id,
-						event.Unchanged, clusterObj)
+				if filtered {
+					klog.V(4).Infof("apply filtered by %s because (%s): %s", filter.Name(), reason, id)
+					taskContext.EventChannel() <- createApplyEvent(id, event.Unchanged, obj)
+					taskContext.CaptureResourceFailure(id)
+					break
 				}
-				taskContext.CaptureResourceFailure(id)
+			}
+			if filtered || filterErr != nil {
```

Comment:
I think it might be clearer to just do continue when these situations happen instead of using break and then this check. It confused me a bit at first.

Reply:
Since there are two `for` loops, a single `continue` does not work. The `break` is to stop the filter loop. The `continue` is to go to the next object.